crates/string_key_newtype/src/lib.rs

```diff
@@ -30,6 +30,16 @@ macro_rules! string_key_newtype {
                 Self(other)
             }
         }
+
+        impl<'de> serde::Deserialize<'de> for $named {
+            fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
```

Comment:
This is neat. If we're making all of the stringkey types deserializable, we can probably make `MagicMutationInfo` contain more appropriate `StringKey` types, e.g. `field: SelectableFieldName` and `path` and `to/`from` could be a new stringkey type `UnvalidatedPath` or something. That should be done as a followup, though.